src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -146,8 +163,23 @@ private Cci.ITypeReference GetStorageStruct(uint size)
             return new ExplicitSizeStruct(size, this, _systemValueType);
         }
 
+        internal Cci.IFieldReference GetMVID(Cci.ITypeReference mvidType)
+        {
+            if (_mvidField == null)
+            {
+                Interlocked.CompareExchange(ref _mvidField, new MvidField("MVID", this, mvidType), null);
+            }
+
+            return _mvidField;
+        }
 
-        // Add a new synthesized method indexed by it's name if the method isn't already present.
+        internal Cci.IMethodDefinition StaticConstructor
```

Comment:
Clearly property is a bad choice for adding a static constructor (setter might have no effect, etc). I think this should be a method GetOrAddStaticConstructor that takes a delegate creating an instance of the constructor and returns back the value of _staticConstructor field.